packages/rpc-transport-http/src/http-transport-for-solana-rpc.ts

```diff
@@ -16,9 +16,27 @@ type Config = Readonly<{
 export function createHttpTransportForSolanaRpc(config: Config): RpcTransport {
     return createHttpTransport({
         ...config,
-        fromJson: (rawResponse: string, request: RpcRequest) =>
-            isSolanaRequest(request) ? parseJsonWithBigInts(rawResponse) : JSON.parse(rawResponse),
+        fromJson: (rawResponse: string, request: RpcRequest) => {
+            if (!isSolanaRequest(request)) return JSON.parse(rawResponse);
+            const response = parseJsonWithBigInts(rawResponse);
+            // Error codes are always numbers.
+            if (isErrorResponse(response)) {
+                return { ...response, error: { ...response.error, code: Number(response.error.code) } };
+            }
```

Comment:
Here we have to downcast the error code from `bigint` to `number` since the response transformer doesn't tackle error responses.